src/modules/producer.js

```diff
@@ -118,7 +119,7 @@ function checkRpc (queue, msg, options) {
 
         if (publishOrSendToQueue.call(this, queue, msg, options)) {
             //defered promise that will resolve when response is received
-            let responsePromise = Promise.defer();
+            let responsePromise =  new Deferred();
```

Comment:
Oh you are right however we need to then set engine entry in package.js to make sure npm install does not work on node < 6.  On that Promise.deffer note here is a good discussion http://stackoverflow.com/questions/34971078/how-to-replace-promise-defer-with-new-promise

I think we need to refactor this piece of code return promises where needed instead of using deferred.